README.md

```diff
@@ -1,7 +1,39 @@
+Margaret is [`go-ssb`](https://github.com/cryptoscope/ssb)'s [append-only](https://en.wikipedia.org/wiki/Append-only)
+log\* provider, and greatly inspired by [flumedb](https://github.com/flumedb/flumedb). Compatible with Go 1.13+.
 
-aka the Log Lady. Helps you to store and interact with append-only logs.
+![margaret the log lady, 1989 edition](https://static.wikia.nocookie.net/twinpeaks/images/6/68/Logladyreplacement.jpg/revision/latest/scale-to-width-down/500?cb=20160906170235)
 
-Heavily inspired by [flumedb](https://github.com/flumedb/flumedb).
+_the project name is inspired by Twin Peaks's character [Margaret](https://twinpeaks.fandom.com/wiki/Margaret_Lanterman) aka **the
+log lady**_
 
-Requires Go version 1.13 or later.
\ No newline at end of file
+Margaret has the following facilities:
+* an append-only log interface `.Append(interface{})`, `.Get(int64)`
+* [queries](https://github.com/cryptoscope/margaret/blob/master/qry.go) `.Query(...QuerySpec)` for retrieving ranges based on sequence numbers e.g. `Query.Gt(int64)`, or limiting the amount of data returned `.Limit(int64)` 
+* a variety of index mechanisms, both for categorizing log entries into buckets and for creating virtual logs (aka sublogs)
+
+### Log storage
+Margaret outputs data according to the [`offset2`](https://godocs.io/go.cryptoscope.co/margaret/offset2) format, which is inspired by (but significantly differs from) [`flumelog-offset`](https://github.com/flumedb/flumelog-offset).
+
+In brief: margaret stores the data of _all logs_ in the three following files:
+* `data` stores the actual data (with a length-prefix before each entry)
+* `ofst` indexes the starting locations for each data entry in `data`
+* `jrnl` an integrity checking mechanism for all three files; a checksum of sorts, [more details](https://github.com/cryptoscope/margaret/blob/master/offset2/log.go#L215)
+
+## More details
+* multilogs, similar to leveldb indexes
+* sublogs (and rxLog/receiveLog/offsetLog and its equivalence to offset.log)
+* indexes
+* queries
+* zeroing out written data
+
+## Components
+Margaret is one of a few components that make the [go implementation of ssb](https://github.com/cryptoscope/ssb/) tick:
+* [`ssb/sbot`](https://github.com/cryptoscope/ssb/) uses margaret for storing each peer's data
+* [`luigi`](https://github.com/cryptoscope/luigi) is used by margaret to pipe data (in the form of [`muxrpc`](https://github.com/cryptoscope/go-muxrpc) responses) from connected scuttlebutt peers into storage on the local machine
+
+
+\* margaret is technically an append-_based_ log, as there is support for
+zeroing out items in the log after they have already been written. Given
```

Comment:
there is zeroing `Null(int64)` and `Replace(int64, []byte)`.

The latter I added to be able to remove content portions of supporting formats (like gabby grove and bamboo) where the header of the message should stay intact.

Reply:
Important caveat is that when using replace, the entry has to be of the same size or smaller then the existing entry (otherwise it would cut into the next entry)

Reply:
added a bit about that, but left out the caveat (too detailed for a readme imo?). maybe that can be put into dev.scuttlebutt.nz?

Reply:
no, let's just put this on the godoc function, if it's not there yet.

Reply:
oh, i see.. since the `offsetLog` type isn't exported none of it is on the godoc page. I'll fix that.